.github/workflows/publish-pr-preview.yml

```diff
@@ -15,59 +15,34 @@ on:
-    steps:
-      - name: Compute secrets availability
-        id: secret_availability
-        env:
-          SURGE_TOKEN: ${{ secrets.SURGE_TOKEN_DOC }}
-        run: |
-          echo "is_SURGE_TOKEN_set: ${{ env.SURGE_TOKEN != '' }}"
-          echo "::set-output name=is_SURGE_TOKEN_set::${{ env.SURGE_TOKEN != '' }}"
-
   build_preview:
-    needs: [check_secrets]
-    # Only build preview for member of the repository
-    if: needs.check_secrets.outputs.is_SURGE_TOKEN_set == 'true'
     runs-on: ubuntu-20.04
+    permissions:
+      pull-requests: write # surge-preview creates or updates PR comments about the deployment status
     env:
-      PR_NUMBER: ${{ github.event.pull_request.number }}
       BONITA_BRANCH: '2021.2'
       BCD_BRANCH: '3.5'
       CLOUD_BRANCH: 'master'
       LABS_BRANCH: 'master'
       TOOLKIT_BRANCH: '1.0'
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-        if: github.event.action != 'closed'
-      - name: Build Setup
-        uses: ./.github/actions/build-setup
-        if: github.event.action != 'closed'
       - name: Compute environment variables
-        if: github.event.action != 'closed'
         run: |
           echo "BRANCH_NAME=$(echo ${GITHUB_HEAD_REF#refs/heads/})" >> $GITHUB_ENV
-          # the surge-preview action generates https://{{repository.owner}}-{{repository.name}}-{{job.name}}-pr-{{pr.number}}.surge.sh
-          repo_owner_and_name=$(echo "${{github.repository}}" | sed 's/\//-/g')
-          echo PREVIEW_URL=https://$repo_owner_and_name-"${{github.job}}"-pr-$PR_NUMBER.surge.sh >> $GITHUB_ENV
-      - name: Publish preview
-        uses: afc163/surge-preview@v1
+      - uses: actions/checkout@v3 # access to the local action
+      - name: Build and publish PR preview
+        uses: ./.github/actions/build-and-publish-pr-preview
         with:
-          surge_token: ${{ secrets.SURGE_TOKEN_DOC }}
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          dist: build/site
-          failOnError: true
-          teardown: 'true'
-          build: |
-            ./build-preview.bash --use-multi-repositories --component-with-branches bonita:"${{ env.BONITA_BRANCH }}" --component-with-branches bcd:"${{ env.BCD_BRANCH }}" --component-with-branches cloud:"${{ env.CLOUD_BRANCH }}" --component-with-branches labs:"${{ env.LABS_BRANCH }}" --component-with-branches test-toolkit:"${{ env.TOOLKIT_BRANCH }}" --pr "${{ env.PR_NUMBER }}" --site-url "${{ env.PREVIEW_URL }}"
-            ls -lh build/site
+          surge-token: ${{ secrets.SURGE_TOKEN_DOC }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          doc-site-branch: ${{ env.BRANCH_NAME }}
+          # '>' Replace newlines with spaces (folded)
+          # '-' No newline at end (strip)
+          build-preview-command: >-
+            ./build-preview.bash --use-multi-repositories
+            --component-with-branches bonita:"${{ env.BONITA_BRANCH }}"
```

Comment:
:heart: nice formatting

Reply:
yes, I discovered this by chance and this makes the configuration easier to read 😄